### ISSUE TITLE
Server matcher failure sometimes makes another test fail on using server pool

### ISSUE THREAD

**Issue by itchyny** [ggriffiniii/httptest#19]
Thanks for maintaining this great crate. I noticed that when matcher fails, another test (which is expected to pass) fails with `Unexpected number of requests for matcher` error. I think this is related using the server pool. Here's a sample test.
```rust
#[cfg(test)]
mod httptest_tests {
    use httptest::{matchers::*, responders::*, Expectation, ServerPool};

    static TEST_SERVER_POOL: ServerPool = ServerPool::new(1);

    #[async_std::test]
    async fn get() {
        let server = TEST_SERVER_POOL.get_server();
        server.expect(
            Expectation::matching(all_of![request::method_path("GET", "/")])
                .respond_with(status_code(200)),
        );
        let client = reqwest::Client::new();
        let resp = client.get(server.url_str("/")).send().await.unwrap();
        assert_eq!(200, resp.status().as_u16());
    }

    #[async_std::test]
    async fn post() {
        let server = TEST_SERVER_POOL.get_server();
        server.expect(
            Expectation::matching(all_of![request::method_path("POST", "/")])
                .respond_with(status_code(200)),
        );
        let client = reqwest::Client::new();
        let resp = client.post(server.url_str("/")).send().await.unwrap();
        assert_eq!(200, resp.status().as_u16());
    }

    #[async_std::test]
    async fn put() {
        let server = TEST_SERVER_POOL.get_server();
        server.expect(
            Expectation::matching(all_of![request::method_path("GET", "/")]) // wrong!
                .respond_with(status_code(200)),
        );
        let client = reqwest::Client::new();
        let resp = client.put(server.url_str("/")).send().await.unwrap();
        assert_eq!(200, resp.status().as_u16());
    }

    #[async_std::test]
    async fn delete() {
        let server = TEST_SERVER_POOL.get_server();
        server.expect(
            Expectation::matching(all_of![request::method_path("DELETE", "/")])
                .respond_with(status_code(200)),
        );
        let client = reqwest::Client::new();
        let resp = client.delete(server.url_str("/")).send().await.unwrap();
        assert_eq!(200, resp.status().as_u16());
    }
}
```

This test sometimes fails only `httptest_tests::put`, but sometimes fails with another test (this is random).
The test `httptest_tests::put` is expected to fail, but `httptest_tests::get` should succeed.

```
 $ cargo test httptest_tests
running 4 tests
test httptest_tests::post ... ok
test httptest_tests::delete ... ok
test httptest_tests::put ... FAILED
test httptest_tests::get ... FAILED

failures:

---- httptest_tests::put stdout ----
thread 'httptest_tests::put' panicked at src/dashboard.rs:441:9:
assertion `left == right` failed
  left: 200
 right: 500
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- httptest_tests::get stdout ----
thread 'httptest_tests::get' panicked at $HOME/.cache/cargo/registry/src/index.crates.io-6f17d22bba15001f/httptest-0.15.4/src/server.rs:79
:17:
Unexpected number of requests for matcher 'AllOf[MethodPath { method: "GET", path: "/" }]'; received 0; expected Exactly(1)


failures:
    httptest_tests::get
    httptest_tests::put

test result: FAILED. 2 passed; 2 failed; 0 ignored; 0 measured; 203 filtered out; finished in 0.01s

error: test failed, to rerun pass `--lib`
```

I expect that tests pass or fail independently even if some test fails with matcher errors.

**Comment by ggriffiniii** [comment]
Thank you for the very detailed report. This should be fixed by c351eef and I've released v0.15.5 that includes the fix. Sorry for the trouble. Let me know if you have any other issues.

**Comment by itchyny** [comment]
Thank you for quick fix! I've confirmed that the issue has been resolved by upgrading the crate.
